src/main/java/agh/ics/oop/OptionParser.java

```diff
@@ -0,0 +1,29 @@
+package agh.ics.oop;
+
+import java.util.Arrays;
+
+public class OptionParser {
+    public static MoveDirection[] parse(String[] directions){
+        MoveDirection[] directionsEnum = new MoveDirection[directions.length];
+        int index = 0;
+        for(String direction : directions) {
+            switch (direction) {
+                case "l", "left" -> directionsEnum[index] = MoveDirection.LEFT;
+                case "f", "forward" -> directionsEnum[index] = MoveDirection.FORWARD;
+                case "r", "right" -> directionsEnum[index] = MoveDirection.RIGHT;
+                case "b", "backwards" -> directionsEnum[index] = MoveDirection.BACKWARD;
+                default -> index--;
+            }
+            index++;
+        }
+        return Arrays.copyOfRange(directionsEnum, 0, index);
+    }
+    public static void run(MoveDirection[] directions, Animal animal){
```

Comment:
z punktu widzenia działania programu to jest ok, ale z punktu widzenia projektu aplikacji to ta metoda niespecjalnie pasuje do klasy, która zajmuje się parsowaniem wejścia, zaburzamy tu tzw. zasadę jednej odpowiedzialności (SRP)